src/ngx_http_lua_time.c

```diff
@@ -272,6 +272,77 @@ ngx_http_lua_ffi_time(void)
 {
     return (long) ngx_time();
 }
+
+
+void
+ngx_http_lua_ffi_update_time(void)
+{
+    ngx_time_update();
+}
+
+
+void
+ngx_http_lua_ffi_today(unsigned char *buf)
+{
+    ngx_tm_t                 tm;
+
+    ngx_gmtime(ngx_time() + ngx_cached_time->gmtoff * 60, &tm);
+
+    ngx_sprintf(buf, "%04d-%02d-%02d", tm.ngx_tm_year, tm.ngx_tm_mon,
+                tm.ngx_tm_mday);
+}
+
+
+void
+ngx_http_lua_ffi_localtime(unsigned char *buf)
+{
+    ngx_tm_t                 tm;
+
+    ngx_gmtime(ngx_time() + ngx_cached_time->gmtoff * 60, &tm);
+
+    ngx_sprintf(buf, "%04d-%02d-%02d %02d:%02d:%02d", tm.ngx_tm_year,
+                tm.ngx_tm_mon, tm.ngx_tm_mday, tm.ngx_tm_hour, tm.ngx_tm_min,
+                tm.ngx_tm_sec);
+}
+
+
+void
+ngx_http_lua_ffi_utctime(unsigned char *buf)
+{
+    ngx_tm_t       tm;
+
+    ngx_gmtime(ngx_time(), &tm);
+
+    ngx_sprintf(buf, "%04d-%02d-%02d %02d:%02d:%02d", tm.ngx_tm_year,
+                tm.ngx_tm_mon, tm.ngx_tm_mday, tm.ngx_tm_hour, tm.ngx_tm_min,
+                tm.ngx_tm_sec);
+}
+
+
+int
+ngx_http_lua_ffi_cookie_time(unsigned char *buf, long t)
+{
+    u_char                              *p;
+
+    p = ngx_http_cookie_time(buf, t);
+    return p - buf;
+}
+
+
+void
+ngx_http_lua_ffi_http_time(unsigned char *buf, long t)
+{
+    ngx_http_time(buf, t);
+}
+
+
+void
+ngx_http_lua_ffi_parse_http_time(const unsigned char *str, size_t len,
+    size_t *time)
```

Comment:
It looks logically wrong to me to use type `size_t` for time values.

Reply:
@agentzh 
I forgot the 32 bit. Fixed in the latest commit.